frontend/src/components/PwReset.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import { Form, Message } from "semantic-ui-react";
+import { useAuthContext } from "../contexts/AuthContext";
+
+const PwReset = () => {
+  const [userInfo, setUserInfo] = useState({
+    email: "",
+    pw: "",
+    pw2: "",
+    code: "",
+  });
+  const [inputErrors, setInputErrors] = useState({
+    email: false,
+    pw: false,
+    pw2: false,
+    code: false,
+  });
+  const [showResetForm, setShowResetForm] = useState(false);
+  const [isValidCode, setIsValidCode] = useState(false);
+  const [message, setMessage] = useState();
+
+  const { sendResetEmail, verifyCode, resetPassword } = useAuthContext();
+
+  const containerStyles = `container mx-auto max-w-sm border-4
+     rounded m-4 p-4`;
+
+  const sendCode = async (e) => {
+    if (userInfo.email === "") {
+      setInputErrors({ ...inputErrors, email: e.target.value });
+      return;
+    } else {
+      // send verification
+      const resp = await sendResetEmail(userInfo.email);
+      setMessage(
+        <Message content="A reset link has been sent to your e-mail." />
+      );
+      setShowResetForm(true);
+      // setMessage(false);
+    }
+  };
+
+  const checkCode = async (e) => {
+    if (userInfo.code === "") {
+      setInputErrors({ ...inputErrors, code: e.target.value });
+      return;
+    } else {
+      // check code before showing reset form
+      try {
+        // await verifyCode(userInfo.code);
+        setIsValidCode(true);
+      } catch (error) {
+        // set error message
+      }
+    }
+  };
+
+  const completeReset = async (e) => {
+    if (userInfo.pw === "" || userInfo.pw !== userInfo.pw2) {
+      if (userInfo.pw === "") {
+        setInputErrors((prevState) => {
+          return { ...prevState, pw: true };
+        });
+      }
+      if (userInfo.pw !== userInfo.pw2) {
+        setInputErrors((prevState) => {
+          return { ...prevState, pw2: true };
+        });
+      }
+      return;
+    } else {
+      // reset password
+      // const resp = await resetPassword(userInfo.code, userInfo.pw);
+      setMessage(<Message content="Password reset successfully." />);
+      // console.log(resp);
+    }
+  };
+
+  const handleInputChange = (e) => {
+    if (e.target.value !== "" && inputErrors[e.target.name] === true) {
+      setInputErrors({ ...inputErrors, [e.target.name]: false });
+    }
+
+    setUserInfo({ ...userInfo, [e.target.name]: e.target.value });
+  };
+
+  return (
+    <div className={containerStyles}>
+      {message}
+      {showResetForm ? (
+        isValidCode ? (
+          <>
```

Comment:
Wrap all of these different views in a separate component with a descriptive name so that the return value will make sense at first glance.

Reply:
I removed those views as they were no longer necessary.